### PR TITLE
chore: Adds requires side of the sdcore_config relation

### PR DESCRIPTION
**Pull request by Gmerold**
# Description

- Adds requires side of the `sdcore_config` relation
- Extracts unit test fixtures to a separate class to reduce code duplications

# Checklist:

- [x] My code follows the [style guidelines](/CONTRIBUTING.md) of this project
- [x] I have performed a self-review of my own code
- [x] I have made corresponding changes to the documentation
- [x] I have added tests that validate the behaviour of the software
- [x] I validated that new and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules
- [x] I have bumped the version of the library

### CODE REVIEW

File: src/charm.py [hunk 1]
```diff
@@ -445,6 +458,13 @@ def _configure_pebble(self, restart: bool = False) -> None:
             logger.info("Restarted container %s", self._service_name)
             return
 
+    def _missing_relations(self) -> list:
```
Comment (gruyaume) [inline src/charm.py:461]:
```suggestion
    def _missing_relations(self) -> List[str]:
```

**Review by gruyaume — APPROVED** [review]
Looks good, minor nit

File: src/charm.py [hunk 2]
```diff
@@ -61,14 +67,16 @@ def __init__(self, *args) -> None:
...
+        self._webui = SdcoreConfigRequires(charm=self, relation_name=SDCORE_CONFIG_RELATION_NAME)
         self.unit.set_ports(SBI_PORT)
-        self._certificates = TLSCertificatesRequiresV3(self, "certificates")
+        self._certificates = TLSCertificatesRequiresV3(self, TLS_RELATION_NAME)
         self._logging = LogForwarder(charm=self, relation_name=LOGGING_RELATION_NAME)
         self.framework.observe(self.on.ausf_pebble_ready, self._configure_ausf)
         self.framework.observe(self.on.update_status, self._configure_ausf)
         self.framework.observe(self.on.fiveg_nrf_relation_joined, self._configure_ausf)
         self.framework.observe(self._nrf_requires.on.nrf_available, self._configure_ausf)
+        self.framework.observe(self._webui.on.webui_url_available, self._configure_ausf)
```
Comment (gatici) [inline src/charm.py:79]: self.framework.observe(self.on.sdcore_config_relation_joined, self._configure_ausf)

I suggest to observe `sdcore_config_relation_joined`  event.

Comment (gatici) [reply src/charm.py:79]: When the relation is joined until webui URL become available, configure method will not be triggered. For the moment, it seems it is affecting our charm state. If we agree on not checking `sdcore_config_relation_joined`, we can also remove `database_relation_joined` and  `fiveg_nrf_relation_joined` observation. They also have no effect on charm state.  Until database_created and nrf_url_available, we can not go forward.

Comment (Gmerold) [reply src/charm.py:80]: Oh, that's right. We can end up with outdated status. 
Handler added.

**Pull request merged**